### PR TITLE
Rever release config & docs

### PR DESCRIPTION
**Pull request by kenodegard**
### Description

As more of us are taking on the role of release manager, we need to publicize the current release process.

xref: https://github.com/conda/infra/issues/541

With this change, we will standardize the `rever.xsh` configuration for `conda`, `conda-build`, and `conda-package-handling`. We will also be syncing out the new `RELEASE.md` docs.

### CODE REVIEW

File: .github/RELEASE.md
Comment (travishathaway) [inline .github/RELEASE.md:87]:
```suggestion
    - If rever finds that any of the authors are not correctly represented in `.authors.yml` it will produce an error. If the author that the error pertains to is:
```

File: .github/RELEASE.md
Comment (travishathaway) [inline .github/RELEASE.md:91]:
```suggestion
    - Once you have successfully run `rever --activities authors` with no errors, review the commit made by rever. This commit will contain updates to one or more of the author files (`.authors.yml`, `.mailmap`, and `AUTHORS.md`). Due to the race condition between `.authors.yml` and `.mailmap`, we want to extract changes made to any of the following keys in `.authors.yml` and commit them separately from the other changes in the rever commit:
```

File: .github/RELEASE.md
Comment (travishathaway) [inline .github/RELEASE.md:100]:
```suggestion
    - Here's a sample run where we undo the commit made by rever in order to commit the changes to `.authors.yml` separately:
```
"rever" right? or did you actually mean "revert"? 🤔

File: .github/RELEASE.md
Comment (travishathaway) [inline .github/RELEASE.md:131]:
```suggestion
      Compare this list with `AUTHORS.md`. If they have any discrepancies, additional modifications to `.authors.yml` is needed, so repeat the above steps as needed.
```

File: .github/RELEASE.md
Comment (travishathaway) [inline .github/RELEASE.md:133]:
```suggestion
    - Once you are pleased with how the author's file looks, we want to undo the rever commit and commit the `.mailmap` changes separately:
```

**Review by travishathaway — APPROVED** [review]
Looks good. I just had a couple grammar/typo things you may want to commit too.

File: .github/RELEASE.md
Comment (beeankha) [inline .github/RELEASE.md:5]: This link doesn't seem to work, it goes to a 404 page (but maybe once this is merged it'll go directly to the file?)

File: .github/RELEASE.md
Comment (beeankha) [inline .github/RELEASE.md:8]: Are these two links currently just placeholders?

File: .github/RELEASE.md
Comment (kenodegard) [reply .github/RELEASE.md:5]: these links aren't expected to work until the file is synced out, its supposed to be a relative link

File: .github/RELEASE.md
Comment (beeankha) [inline .github/RELEASE.md:48]: Nit: 
```suggestion
### 2. Ensure `rever.xsh` and `news/TEMPLATE` are up to date.
```

File: .github/RELEASE.md
Comment (beeankha) [inline .github/RELEASE.md:55]:
```suggestion
Currently, there are only 2 activities we use `rever` for, (1) aggregating the authors and (2) updating the changelog. Aggregating the authors can be an error-prone process and also suffers from builtin race conditions (i.e. to generate an updated `.authors.yml` we need an updated `.mailmap` but to have an updated `.mailmap` we need an updated `.authors.yml`). This is why the following steps are very heavy-handed (and potentially repetitive) in running `rever` commands, undoing commits, squashing/reordering commits, etc.
```

File: .github/RELEASE.md
Comment (kenodegard) [reply .github/RELEASE.md:5]: **but** after further reflection I believe the relative URL will still be wrong, when we view this file on GH it'll be at, e.g., https://github.com/conda/conda/blob/main/RELEASE.md so we want:

```suggestion
[epic template]: ../../issues/new?assignees=&labels=epic&template=epic.yml
```

File: .github/RELEASE.md
Comment (kenodegard) [reply .github/RELEASE.md:8]: **but** after further reflection I believe the relative URL will still be wrong, when we view this file on GH it'll be at, e.g., https://github.com/conda/conda/blob/main/RELEASE.md so we want:

```suggestion
[compare]: ../../compare
[release notes]: ../../releases/new
```

File: .github/RELEASE.md
Comment (beeankha) [inline .github/RELEASE.md:159]:
```suggestion
4. Review news snippets (ensure they are all using the correct Markdown format, **not** reStructuredText) and add additional snippets for undocumented PRs/changes as necessary.
```

File: .github/RELEASE.md
Comment (beeankha) [inline .github/RELEASE.md:172]: Just a nitpick, I found the first-person voice here to not quite fit in to the style of the rest of the document (this is an optional "take it or leave it" suggestion):
```suggestion
    - You can utilize [GitHub's compare view][compare] to review what changes are to be included in this release.
```

**Review by travishathaway — APPROVED** [review]

File: .github/RELEASE.md
Comment (jezdez) [inline .github/RELEASE.md:5]: Smart!

File: .github/RELEASE.md
Comment (jezdez) [inline .github/RELEASE.md:39]: I wonder if we should add an item here like "interact with conda-forge maintainers" or something to make clear that bumping the version in the cf recipe alone doesn't mean it's released.

File: .github/RELEASE.md
Comment (jezdez) [inline .github/RELEASE.md:42]: Would be cool if we'd have a @conda/comms team or something to automatically ping here.

**Review by jezdez — APPROVED** [review]
LGTM, only minor comments! 🙇

File: .github/RELEASE.md
Comment (kenodegard) [reply .github/RELEASE.md:39]: hm ok I'll keep this in mind and workshop it in https://github.com/conda/conda/issues/11802 and https://github.com/conda/conda-package-handling/issues/120, we can add whatever we decide on in a future PR?

File: .github/RELEASE.md
Comment (kenodegard) [reply .github/RELEASE.md:42]: True, this would be something that we want to configure with Terraform?

File: .github/RELEASE.md
Comment (kenodegard) [reply .github/RELEASE.md:39]: https://github.com/conda/infra/issues/602

Comment (jezdez) [reply .github/RELEASE.md:39]: Sounds good, thank you!

File: .github/RELEASE.md
Comment (jezdez) [reply .github/RELEASE.md:42]: Once we have that, yeah. I can find that branch and push to a infra repo branch next week.

**Pull request merged**
